tests/admin_checks/tests.py

```diff
@@ -105,7 +117,29 @@ class CustomAdminSite(admin.AdminSite):
             self.assertEqual(errors, expected)
         finally:
             custom_site.unregister(Song)
-            admin.sites.system_check_errors = []
+
+    @override_settings(DEBUG=True)
+    def test_allows_checks_relying_on_other_modeladmins(self):
+        class MyBookAdmin(admin.ModelAdmin):
+            def check(self, **kwargs):
+                errors = super(MyBookAdmin, self).check(**kwargs)
+                author_admin = self.admin_site._registry.get(Author)
+                if author_admin is None:
+                    errors.append('AuthorAdmin missing!')
+                return errors
+
+        class MyAuthorAdmin(admin.ModelAdmin):
+            pass
+
+        admin.site.register(Book, MyBookAdmin)
+        admin.site.register(Author, MyAuthorAdmin)
+
+        errors = checks.run_checks()
```

Comment:
This could work. I could make this modification for this test but I think that by calling `checks.run_checks()` you also test that the admin checks are properly registered into the checks registry.